R/sdk.R

```diff
@@ -0,0 +1,99 @@
+#' Install the Google Cloud SDK
+#'
+#' Installs the Google Cloud SDK which enables CloudML operations.
+#'
+#' @export
+sdk_install <- function() {
+  if (dir.exists(sdk_dir())) {
+    message("SDK already installed.")
+    return(invisible(NULL))
+  }
+
+  if (Sys.info()["sysname"] != "Darwin")
+    stop("Currently, only OS X installation is supported.")
+
+  cloudsdk_url <- "https://dl.google.com/dl/cloudsdk/channels/rapid/downloads/"
+  cloudsdk_tar <- "google-cloud-sdk-178.0.0-darwin-x86_64.tar.gz"
+
+  download_dir <- tempdir()
+  download_file <- file.path(download_dir, cloudsdk_tar)
+
+  if (!dir.exists(download_dir)) dir.create(download_dir)
+
+  download.file(
+    file.path(cloudsdk_url, cloudsdk_tar),
+    download_file
+  )
+
+  extract_dir <- "~/"
+  untar(tarfile = download_file,
+        exdir = extract_dir,
+        tar = "internal")
+}
+
+#' Retrieve Google Cloud SDK Path
+#'
+#' Retrieves the path to the Google Cloud SDK.
+#'
+#' @export
+sdk_dir <- function() {
+  "~/google-cloud-sdk"
+}
+
+#' Trigger Google Cloud SDK Command
+#'
+#' Triggers a Google Cloud SDK command.
+#'
+#' @param ... Parameters to use with 'gcloud'.
+#'
+#' @export
+sdk_command <- function(...) {
+  params <- list(...)
+  gcloud <- file.path(sdk_dir(), "bin", "gcloud")
+
+  gexec(
+    normalizePath(gcloud),
+    params
+  )
+}
+
+#' Login to Google Cloud SDK
+#'
+#' Login to Google Cloud SDK to enable CloudML operations.
+#'
+#' @export
+sdk_login <- function() {
+  sdk_command("auth", "application-default", "login")
+}
+
+#' Config the Google Cloud SDK
+#'
+#' Config the Google Cloud SDK to enable CloudML operations.
+#'
+#' @param account The account to use in subsequent Google Cloud SDK operations.
+#'   Usually the email associated to the account.
+#' @param project The project to use in subsequent Google Cloud SDK operations.
+#'
+#' @export
+sdk_config <- function(account, project) {
```

Comment:
These APIs will set state on whatever the current, active configuration is. Do we also need APIs / do we need to teach users how to activate different `gcloud` configurations?

Reply:
In addition, previously, we were advertising that this state should be set within e.g. `cloudml.yml`. Now users have two different mechanisms for setting an account + project, where `cloudml.yml` is a 'project-specific' configuration file and this is a 'global' configuration.

Reply:
In other words, I think it's worth expanding the documentation here just to make the distinction clear to users.

Reply:
I ended up reverting `gcloud_account/project`, mostly since I wasn't fully aware `cloudml.yml` was the path to go, so now we encourage this path and we don't add extra functions to set/get account/project.